ledger-test-tool/suites/src/main/scala/com/daml/ledger/api/testtool/suites/v1_8/PartyManagementServiceIT.scala

```diff
@@ -20,13 +21,311 @@ import scalaz.Tag
 import scalaz.syntax.tag.ToTagOps
 import java.util.regex.Pattern
 
+import com.daml.ledger.api.v1.admin.identity_provider_config_service.DeleteIdentityProviderConfigRequest
 import com.daml.ledger.api.v1.admin.object_meta.ObjectMeta
 import com.daml.ledger.client.binding.Primitive
 
 import scala.util.Random
 
 final class PartyManagementServiceIT extends PartyManagementITBase {
 
+  test(
+    "PMUpdatingPartyIdentityProviderNonDefaultIdps",
+    "Test reassigning party to a different idp using non default idps",
+    allocate(NoParties),
+  )(implicit ec => { case Participants(Participant(ledger)) =>
+    val idpId1 = ledger.nextIdentityProviderId()
+    val idpId2 = ledger.nextIdentityProviderId()
+    for {
+      _ <- ledger.createIdentityProviderConfig(identityProviderId = idpId1)
+      _ <- ledger.createIdentityProviderConfig(identityProviderId = idpId2)
+      party <- ledger.allocateParty(identityProviderId = Some(idpId1)).map(_.unwrap)
+      get1 <- ledger.getParties(
+        GetPartiesRequest(parties = Seq(party), identityProviderId = idpId1)
+      )
+      // Update party's idp id
+      _ <- ledger.updatePartyIdentityProviderId(
+        UpdatePartyIdentityProviderRequest(
+          party = party,
+          sourceIdentityProviderId = idpId1,
+          targetIdentityProviderId = idpId2,
+        )
+      )
+      get2 <- ledger.getParties(
```

Comment:
it would make sense to check that the party cannot be found anymore in the idp1

Reply:
Ok